plugins/module_utils/network/sonic/config/poe/poe.py

```diff
@@ -0,0 +1,741 @@
+)
+
+
+class Poe(ConfigBase):
+    """
+    The sonic_poe class
+    """
+
+    diff_keys = [
+        {"cards": {"card_id": ""}},
+        {"interfaces": {"name": ""}}
+    ]
+
+    poe_root_uri = "data/openconfig-poe:poe"
+    interfaces_root_uri = "data/openconfig-interfaces:interfaces"
+
+    poe_setting_prefix = "openconfig-if-poe-ext:"
+    '''prefix for interface's PoE settings'''
+
+    gather_subset = [
+        '!all',
+        '!min',
+    ]
+
+    gather_network_resources = [
+        'poe',
+    ]
+
+    # tracks keys and values still entirely unsupported by platforms
+    # unsupported_values_dict should have same nesting scheme as argspec.
+    # if key is unsupported specify blank value, otherwise fill in with list of values or singular value that isn't supported
+    # keep key in if it has sub sections that have unsupported values
+    unsupported_values = {
+        "cards": "",
+        "global": {
+            "auto_reset": "",
+            "power_mgmt_model": [
+                'dynamic-priority',
+                'static',
+                'static-priority'
+            ],
+            "usage_threshold": "",
+        },
+        "interfaces": {
+            "priority": ['medium'],
+            "detection": [
+                '2pt-dot3af',
+                '2pt-dot3af+legacy',
+                '4pt-dot3af',
+                '4pt-dot3af+legacy',
+                'legacy'
+            ],
+            "power_up_mode": "",
+            "power_pairs": "",
+            "power_limit_type": "",
+            "power_limit": "",
+            "high_power": "",
+            "disconnect_type": "",
+            "four_pair": "",
+            "use_spare_pair": "",
+            "power_classification": ""
+        }
+    }
+
+    def __init__(self, module):
+        super(Poe, self).__init__(module)
+
+    def get_poe_facts(self):
+        """ Get the 'facts' (the current configuration)
+
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts(self._module).get_facts(self.gather_subset, self.gather_network_resources)
+        poe_facts = facts['ansible_network_resources'].get('poe')
+        if not poe_facts:
+            return []
+        return poe_facts
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = list()
+
+        existing_poe_facts = self.get_poe_facts()
+        commands, requests = self.set_config(existing_poe_facts)
+
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.errno)
+                result['changed'] = True
```

Comment:
Sorry...

My previous comment about this instruction was incorrect. I realized later that to show the correct simulated result in 'check' mode, 'changed' does need to be set to 'true' if the outer condition ('if commands and len(requests) > 0:') is met, so please outdent this instruction back to where it was before so it will still execute if 'check' mode is specified and the outer condition is met. I am posting another revision request related to this one.